### DISCUSSION TITLE
A GitHub readme Badge for the OpenModelica compatibility score

### DISCUSSION THREAD

**Discussion by Maghnie** [OpenModelica/OpenModelica#10368] [Ideas]
Since the awesome work was implemented for [checking the compatibilities of open-source Modelica libraries with OpenModelica](https://libraries.openmodelica.org/branches/master/), it would be nice and useful for motivating open-source development if Modelica libraries can show (off) their OpenModelica compatibility score on their readme, e.g. [through a badge](https://github.com/Naereen/badges).

Currently, we're considering realizing this feature by scraping the relevant stats from [the OpenModelica Webpage](https://libraries.openmodelica.org/branches/master/) where they're all listed. This way is not optimal though, because we would have to rely on the webpage structure not changing or else it would break our method.

Or does anyone have a more practical idea to implement the OpenModelica score badge?

**Comment by casella** [comment]
@Maghnie thanks for your efforts!

You can find more information about the reporting of OpenModelica's CI here: https://github.com/OpenModelica/OpenModelicaLibraryTesting

We have no plans of changing the structure of https://libraries.openmodelica.org/branches/master/ for the foreseeable future. It just serves us well as it is. We'll just add more libraries over time.

**Reply by AnHeuermann** [reply]
I think @Maghnie is asking for something like the badges we produce when testing OMJulia:

[![Test](https://github.com/OpenModelica/OMJulia.jl/actions/workflows/Test.yml/badge.svg)](https://github.com/OpenModelica/OMJulia.jl/actions/workflows/Test.yml)

In the end this is no more than an svg image at some location, linked from a README (or command in this case).
We have scripts for the library testing. If you want to add a small script to generate the picture from the information we have in the Python scripts it shouldn't be hard to upload it to e.g. `https://libraries.openmodelica.org/branches/master/AixLib/files/badge.svg`.
Please open a feature request in https://github.com/OpenModelica/OpenModelicaLibraryTesting if want to provide such a pull request.

**Comment by casella** [comment]
Just one comment: https://libraries.openmodelica.org/branches/master/ contains directories that are now stale. Maybe we should just delete them at some point. You should probably check the timestamps of files within that directory, and discard them if they are more than, say, 30 days old, because that means the testing has been discontinued.
